docs/source/docs/configuration.md

```diff
@@ -2,19 +2,459 @@
+
+#### Image
+
+- `weight`: float type, weight of individual loss element in total loss function.
+- `name`: string type, one of "lncc", "ssd" or "gmi". EG.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    dissimilarity:
+      image:
+        name: "lncc" # other options include "lncc", "ssd" and "gmi", for local normalised cross correlation,
+        weight: 0.1
+```
+
+#### Label
+
+- `weight`: float type, weight of individual loss element in total loss function.
+- `name`: string type, "multi_scale" or "single_scale", defines the resolution of
+  kernels to use with the loss.
+  - The same string should be passed as a field to the dictionary, with the following
+    additional arguments under each field:
+  - `multi_scale`:
+    - `loss_type`: string type, name of loss to use. Options currently include "dice",
+      "cross-entropy", "mean-squared", "generalised_dice" and "jaccard".
+    - `loss_scales`: list of ints, what
+  - `single_scale`:
+    - `loss_type`: name of loss to use. Options currently include "dice",
+      "cross-entropy", "mean-squared", "generalised_dice" and "jaccard".
+
+EG.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    dissimilarity:
+      label:
+        weight: 1.0
+        name: "multi_scale" # "multi_scale" or "single_scale"
+        multi_scale:
+          loss_type: "dice" # options include "dice", "cross-entropy", "mean-squared", "generalised_dice" and "jaccard"
+          loss_scales: [0, 1, 2, 4, 8, 16, 32]
+```
+
+#### Regularization
+
+The regularization section configures the losses for the DDF. To instantiate this part
+of the loss, pass "regularization" into the config file as a field.
+
+- `weight`: float type, weight of the regularization loss.
+- `energy_type`: string type, the type of deformation energy to compute. Options include
+  "bending", "gradient-l1", and "gradient-l2"
+
+EG.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    regularization:
+      weight: 0.5 # weight of regularization loss
+      energy_type: "bending" # options include "bending", "gradient-l1" and "gradient-l2"
+```
+
+### Optimizer - required
+
+The optimizer can be defined by using a `name` and then passing optimizer specific
+arguments with the same name. All optimizers can use the `learning_rate` argument.
```

Comment:
same as backbone, except the `name` all others will be passed to the class directly

Reply:
oh.. seems like i haven't do this refactoring yet. forget this thread